packages/dashboard-frontend/src/containers/__tests__/FactoryLoader.spec.tsx

```diff
@@ -181,6 +181,78 @@ describe('Factory Loader container', () => {
   });
 
   describe('with prebuilt resources', () => {
+    it('should not try to create a new workspace and show an error if prebuilt resources are empty', async () => {
+      const store = new FakeStoreBuilder()
+        .withWorkspacesSettings({ 'che.devworkspaces.enabled': 'true' } as che.WorkspaceSettings)
+        .withInfrastructureNamespace([{ name: namespace, attributes: { phase: 'Active' } }], false)
+        .build();
+      const props = getMockRouterProps(ROUTE.LOAD_FACTORY_URL, { url: 'http://test-location' });
+      props.location.search += '&devWorkspace=devWorkspace.yaml';
+
+      (mockAxios.get as jest.Mock).mockResolvedValueOnce({ data: '' });
+
+      render(
+        <Provider store={store}>
+          <FactoryLoaderContainer {...props} />
+        </Provider>,
+      );
+
+      const elementHasError = screen.getByTestId('factory-loader-has-error');
+      await waitFor(() => expect(elementHasError.innerHTML).toEqual('true'));
+      await waitFor(() => expect(createWorkspaceFromResourcesMock).not.toHaveBeenCalled());
+    });
+
+    it('should not try to create a new workspace and show an error if prebuilt resources does not include a Template', async () => {
+      const store = new FakeStoreBuilder()
+        .withWorkspacesSettings({ 'che.devworkspaces.enabled': 'true' } as che.WorkspaceSettings)
+        .withInfrastructureNamespace([{ name: namespace, attributes: { phase: 'Active' } }], false)
+        .build();
+      const props = getMockRouterProps(ROUTE.LOAD_FACTORY_URL, { url: 'http://test-location' });
+      props.location.search += '&devWorkspace=devWorkspace.yaml';
+
+      const yamlContent = `apiVersion: workspace.devfile.io/v1alpha2
+kind: DevWorkspace
+metadata:
+  name: project
+  namespace: test-dev`;
+      (mockAxios.get as jest.Mock).mockResolvedValueOnce({ data: yamlContent });
+
+      render(
+        <Provider store={store}>
+          <FactoryLoaderContainer {...props} />
+        </Provider>,
+      );
+
+      const elementHasError = screen.getByTestId('factory-loader-has-error');
+      await waitFor(() => expect(elementHasError.innerHTML).toEqual('true'));
+      await waitFor(() => expect(createWorkspaceFromResourcesMock).not.toHaveBeenCalled());
```

Comment:
it makes sense to wait for the mock to be called but not vice versa
```suggestion
      expect(createWorkspaceFromResourcesMock).not.toHaveBeenCalled();
```

Reply:
It makes sense. Thanks. I have fixed it.